### PR TITLE
Respect ini session.use_strict_mode

### PR DESCRIPTION
**Pull request by rhertogh**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️/❌Only if Session::openSession() and/or Session::writeSession() are overwritten.
| Tests pass?   | ❌
| Fixed issues  | Mongodb implementation for https://github.com/yiisoft/yii2/pull/18247

### CODE REVIEW

**Comment by rhertogh** [comment]
@samdark I suggest increasing the minor version to 2.2.0 because the Yii2 version constraint is increased significantly.

File: composer.json [hunk 1]
```diff
@@ -18,11 +18,12 @@
         }
     ],
     "require": {
-        "yiisoft/yii2": "~2.0.14",
+        "yiisoft/yii2": "~2.0.38",
```
Comment (samdark) [inline composer.json:21]:
```suggestion
        "yiisoft/yii2": "~2.0.39",
```

File: composer.json [hunk 2]
```diff
@@ -18,11 +18,12 @@
...
         "ext-mongodb": ">=1.0.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "4.8.27|~5.7.21|^6.2"
+        "phpunit/phpunit": "4.8.27|~5.7.21|^6.2",
+        "yiisoft/yii2-dev": "~2.0.38"
```
Comment (samdark) [inline composer.json:26]: Why do we need this dependency?

File: src/Collection.php [hunk 3]
```diff
@@ -241,6 +241,16 @@ public function findOne($condition = [], $fields = [], $options = [])
         return empty($rows) ? null : current($rows);
     }
 
+    /**
+     * Returns if a document exists
+     * @param array $condition  query condition
+     * @return bool
```
Comment (samdark) [inline src/Collection.php:247]: Need `@since 2.0.39` That's new method, right?

File: src/Collection.php [hunk 4]
```diff
@@ -241,6 +241,16 @@ public function findOne($condition = [], $fields = [], $options = [])
         return empty($rows) ? null : current($rows);
     }
 
+    /**
+     * Returns if a document exists
+     * @param array $condition  query condition
```
Comment (samdark) [inline src/Collection.php:246]:
```suggestion
     * @param array $condition Query condition.
```

File: src/Collection.php [hunk 5]
```diff
@@ -241,6 +241,16 @@ public function findOne($condition = [], $fields = [], $options = [])
         return empty($rows) ? null : current($rows);
     }
 
+    /**
+     * Returns if a document exists
```
Comment (samdark) [inline src/Collection.php:245]:
```suggestion
     * Returns if a document exists.
```

File: src/Collection.php [hunk 6]
```diff
@@ -241,6 +241,16 @@ public function findOne($condition = [], $fields = [], $options = [])
...
+     * @param array $condition  query condition
+     * @return bool
+     */
+    public function documentExists($condition = [])
+    {
+        return static::findOne($condition, ['_id' => 1]) === null ? false : true;
```
Comment (samdark) [inline src/Collection.php:251]:
```suggestion
        return static::findOne($condition, ['_id' => 1]) !== null;
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (rhertogh) [reply composer.json]: I might have used it for testing, but can't think of any reason at the moment -> removed.

File: src/Collection.php (hunk 3 above)
Comment (rhertogh) [reply src/Collection.php:247]: jep

**Comment by rhertogh** [comment]
@samdark Feedback has been applied

**Comment by samdark** [comment]
@rhertogh would you please into tests failure? Thanks.

**Comment by samdark** [comment]
Oh, dreaded `@runInSeparateProcess` :)

**Comment by samdark** [comment]
Thank you!

**Pull request merged**